cdm/src/main/java/ucar/nc2/ft2/coverage/simpgeometry/SimpleGeometryKitten.java

```diff
@@ -0,0 +1,89 @@
+package ucar.nc2.ft2.coverage.simpgeometry;
+
+import java.io.IOException;
+
+import ucar.ma2.Array;
+import ucar.nc2.Variable;
+import ucar.nc2.constants.CF;
+import ucar.nc2.ft2.coverage.simpgeometry.*;
+
+/**
+ * A cat (the animal) trained in finding Simple Geometry.
+ * 
+ * A Simple Geometry Kitten can go find the beginning and end indicies of
+ * a simple geometry X and Y within a variable. But first the Kitten needs a few variables to find it in.
+ * The kitten remembers previous adventures so if the kitten is tasked to find a Simple Geometry close to one before it
+ * will find it faster.
+ * 
+ * @author wchen@usgs.gov
+ *
+ */
+public class SimpleGeometryKitten {

```

Comment:
I think this name is cute, but I can't tell, is the Kitten a performance optimization, or a necessity? If it is a performance optimization, can it be removed? If it is a necessity, can it be given a name that makes its purpose more obvious?

Reply:
Whether or not it applies here, it's important to know about the following wisdom from the CS guru Donald Knuth, or as I like to call him, Donald "How is your name on so many important papers?" Knuth.

> Programmers waste enormous amounts of time thinking about, or worrying about, the speed of noncritical parts of their programs, and these attempts at efficiency actually have a strong negative impact when debugging and maintenance are considered. We should forget about small efficiencies, say about 97% of the time: premature optimization is the root of all evil. Yet we should not pass up our opportunities in that critical 3%

I'm not as dramatic as Knuth, so I wouldn't call anything "evil". I am wondering if we can do without this in the short term.

Reply:
Sure sounds good. I'll change it to "SimpleGeometryIndexer" which is more descriptive

Reply:
Though I will remark that it was a reference to how "Dogs" are used as search engine mascots but instead this is a cat.

Reply:
Or actually "SimpleGeometryIndexFinder"

Reply:
Concerning the premature optimization thing, the way it is implemented in the (formerly what was) the Kitty is, is necessary for the common case, which is taking in several sequential geometries. Having it would seem to have a decent impact on performance here.

Reply:
Would the absence of this optimization cause a common use case to take more than 5 minutes to execute?